frontend/src/contexts/HostContext/HostContext.spec.js

```diff
@@ -141,136 +141,342 @@ describe('Context', () => {
+        act(() => {
+          eventHandlers.message({
+            event: 'player-connected',
+            payload: { playerId: 'TEST', playerName: 'TESTER' },
+          });
 
-      render(
-        <HostProvider>
-          <TestComponent />
-        </HostProvider>,
-      );
+          eventHandlers.message({
+            event: 'player-connected',
+            payload: { playerId: 'TEST3', playerName: 'TESTER2' },
+          });
+        });
 
-      expect(screen.queryAllByTestId('player-staging').length).toBe(0);
+        expect(screen.queryAllByTestId('player-staging').length).toBe(0);
+        expect(screen.getAllByTestId('player-ids').length).toBe(2);
 
-      act(() => {
-        eventHandlers.message({
-          event: 'player-connected',
-          payload: { playerId: 'TEST' },
+        act(() => {
+          eventHandlers.message({
+            event: 'player-connected',
+            payload: { id: 'TEST2', playerName: 'TESTER2' },
+          });
         });
+
+        expect(screen.queryAllByTestId('player-staging').length).toBe(0);
+        expect(screen.getAllByTestId('player-ids').length).toBe(2);
       });
 
-      expect(socketInstance.sendMessage).toHaveBeenCalledWith({
-        event: 'update',
-        recipients: ['TEST'],
-        payload: {
-          gameState: 'connected',
-          message: {
-            big: 'Attempting to join lobby',
-            small: 'Please wait',
+      it('tells the middleware to tell the player to wait while they join the lobby', () => {
+        const TestComponent = () => {
+          const { state, dispatch } = useContext(HostContext);
+
+          useEffect(() => {
+            dispatch({
+              type: 'CREATE_LOBBY',
+              payload: {},
+            });
+          }, []);
+
+          return (
+            <>
+              <div data-testid="game-state">{state.gameState}</div>
+              <div data-testid="lobby-id">{state.lobbyID}</div>
+
+              <div>
+                {state.newPlayerStaging.map((player) => (
+                  <span key={player} data-testid="player-staging" />
+                ))}
+              </div>
+            </>
+          );
+        };
+
+        const { eventHandlers } = setupEmitterMocks();
+
+        render(
+          <HostProvider>
+            <TestComponent />
+          </HostProvider>,
+        );
+
+        expect(screen.queryAllByTestId('player-staging').length).toBe(0);
+
+        act(() => {
+          eventHandlers.message({
+            event: 'player-connected',
+            payload: { playerId: 'TEST', playerName: 'TESTER' },
+          });
+        });
+
+        expect(socketInstance.sendMessage).toHaveBeenCalledWith({
+          event: 'update',
+          recipients: ['TEST'],
+          payload: {
+            gameState: 'connected',
+            message: {
+              big: 'Attempting to join lobby',
+              small: 'Please wait',
+            },
           },
-        },
+        });
```

Comment:
Note to make reviewing easier: I didn't change anything here other than moving it into another describe block to group all of these (now many) `player-connected` related tests together.